synedrion/src/cggmp21/protocols/key_refresh.rs

```diff
@@ -60,10 +60,18 @@ enum KeyRefreshErrorEnum<P: SchemeParams> {
         mu: Randomizer<P::Paillier>,
     },
 }
-
+// TODO: The requirement that this be Serialize/Deserialize is what causes
```

Comment:
@fjarri Do you have any advice for me here? [This](https://github.com/entropyxyz/synedrion/pull/120/commits/2af6086b40a6a77927f67cf9a139d5e005406675#diff-94f155822ad48c350d153fcc6840853dc77c75ec9a12a008e2d098913437dbf7R284) along with the requirement of being `Serialize`&`Deserialize` here causes all these bounds to be required and what's worse is they spread out to many places as you can see in this commit. :/

Reply:
I don't think much can be done here, it's caused by weird `serde`'s heuristic for deriving bounds - see #10 and the link to the `serde` issue therein. I find it strange that you have to explicitly specify the bound on `Odd<Uint>` though - is it really required?

Reply:
Sadly yes, but I think it's simply that `crypto-bigint` is missing the `Serialize`/`Deserialize` impls for `Odd`. It does provide them for `NonZero`, `Wrapping`, `Uint`, `Limb` etc so I don't see why not provide one for `Odd`.

FWIW I ran into similar issues with serde and even tried to dig deeper into the source code. I found it to be a very difficult code base to work with and did not make progress.

Do you think there's any possibility that we can get around having so many types be `Serialize`/`Deserialize` if we provide manual impls for the exact types that we need? In other words: is this a case where we're actually only need to serialize the round messages that are sent on the wire? Or are there so many types of serialization tasks and types that we really need the whole type hierarchy to be serializable?

Reply:
We need to serialize messages and also entities (key shares, aux data and so on), which contain in them all this hierarchy, so I am not sure if writing explicit impls will be any easier. 

Would it help adding the `Serialize` impl for `Odd` in `crypto-bigint`? Seems like an omission there.